src/groupByKey.ts

```diff
@@ -1,7 +1,18 @@
-type GroupsMap<T> = {
-  [key: string]: T[];
-};
-
-export function groupByKey(items, key) {
+export function groupByKey<T>(items: T[], key: keyof T): object {
   // write code here;
+  const result: any = {};
+
+  items.forEach((a: T) => {
+    const variable: any = a[key];
```

Comment:
Same here. Also, try not to use any. Set the correct type instead

Reply:
In this place, "Result" as object, adding inside any type of element(number, string etc) example sort by age:
{
...
24: [
  {
      "name": "Diana",
      "surname": "Dorsey",
      "age": 24,
      "married": false,
      "grades": [3, 3, 4, 5, 4, 3, 5, 5]
    },
    {
      "name": "Pamela",
      "surname": "Casillas",
      "age": 24,
      "married": false,
      "grades": [4, 5, 4, 5, 5, 4, 3, 2, 3, 3, 3, 2]
    }
],
...
},
"Variable" can have any names taken from element of object[key]:
"Diana" or 24 or [3, 3, 4, 5, 4, 3, 5, 5]

In condition (ReadME), I dont see limit of type for result.

Reply:
Remake for string.